apps/crn-server/src/data-providers/tutorials.data-provider.ts

```diff
@@ -0,0 +1,30 @@
+import { TutorialsDataObject } from '@asap-hub/model';
+import { SquidexGraphqlClient } from '@asap-hub/squidex';
+import {
+  FetchTutorialsQuery,
+  FetchTutorialsQueryVariables,
+} from '../autogenerated-gql/graphql';
+import { parseGraphQLTutorials } from '../entities';
+import { FETCH_TUTORIAL } from '../queries/tutorials.queries';
+
+export interface TutorialsDataProvider {
+  fetchById(id: string): Promise<TutorialsDataObject | null>;
+}
+export class TutorialsSquidexDataProvider implements TutorialsDataProvider {
+  constructor(private squidexGraphlClient: SquidexGraphqlClient) {}
+  async fetchById(id: string): Promise<TutorialsDataObject | null> {
+    const { findTutorialsContent } = await this.queryFetchByIdData(id);
+
+    if (!findTutorialsContent) {
+      return null;
+    }
+    return parseGraphQLTutorials(findTutorialsContent);
+  }
+
+  private async queryFetchByIdData(id: string) {
```

Comment:
not sure if this extraction is necessary at all?

Reply:
Thanks, it seems cleaner not extracting. It was committed here https://github.com/yldio/asap-hub/pull/2143/commits/a13f9292567850033f7aa025543e577bb831a44c